### ISSUE TITLE
[1.0alpha] Variant order is not respected

### ISSUE THREAD

**Issue by agos** [Shopify/js-buy-sdk#404]
When doing a query, either through the SDK or directly via GrapQL, that involves a Variant Connection (for example, calling `fetchAllProducts()`), the are returned in ascending id order, even if a custom ordering was set in the backoffice.

### Steps to reproduce:

- Create a product with Variants. Add two options with two possible values each to make it easy to see.
- Set a custom ordering of the options by using "Reorder variants"
- Verify that the Variants appear in the selected order in the backoffice
- Use the SDK to call `fetchAllProducts` or another call which will include products and variants, such as fetchCollectionWithProducts()

### Expected result:
- The variants appear in the same order as they appeared in the backoffice

### Actual result:
- The variants are ordered by ascending id

**Comment by minasmart** [comment]
You're right. We should add a `sortKey` argument to the variant connection on field on `Product`. Thanks for pointing this out. I'm adding it to our internal API backlog.

**Comment by alexdupre** [comment]
Any update on this? It's quite important to be able to sort a collection or at least to get the configured collection sorting.

**Comment by minasmart** [comment]
Turns out it's not a non-trivial add. It's in our backlog, but we're just working on prioritization.

**Comment by rebeccajfriedman** [comment]
Closing as stale.
